packages/flutter/test/material/date_picker_theme_test.dart

```diff
@@ -776,17 +775,25 @@ void main() {
       ),
     );
 
+    MaterialInkController findDayGridMaterial(WidgetTester tester) {
+      // All days are painted on the same Material widget.
+      // Use an arbitrary day to find this Material.
+      return Material.of(tester.element(find.text('17')));
+    }
```

Comment:
### 😍😍

This is wonderful, thank you

Reply:
Thanks!
It took me several hours to figure out that there is not one Material for each day. 
This was mainly because existing tests were carefully using the day they are interested in to look up the Material. It was ok, until I spent hours fixing one test where the circle painted for another day collides with the one I was interested in.
I created this utility function to save time for future contributors who will work on the date picker.

Reply:
I have some mixed feelings about the Ink widget—it seems like it'd help performance by consolidating a bunch of widgets into a single ink controller, but it also has the tendency to [introduce jank](https://main-api.flutter.dev/flutter/material/Ink-class.html#limitations), and while it appears to work very similar to DecoratedBox, some things are super confusing for anyone who doesn't have a solid grasp of how painting on a Material works.